typescript/widget-store/src/shop.ts

```diff
@@ -0,0 +1,199 @@
+import { DBOS } from '@dbos-inc/dbos-sdk';
+import { KnexDataSource } from '@dbos-inc/knex-datasource';
+
+export enum OrderStatus {
+  PENDING = 0,
+  DISPATCHED = 1,
+  PAID = 2,
+  CANCELLED = -1,
+}
+
+export interface Product {
+  product_id: number;
+  product: string;
+  description: string;
+  inventory: number;
+  price: number;
+}
+
+export interface Order {
+  order_id: number;
+  order_status: number;
+  last_update_time: Date;
+  product_id: number;
+  progress_remaining: number;
+}
+
+export interface OrderWithProduct {
+  order_id: number;
+  order_status: number;
+  last_update_time: Date;
+  product_id: number;
+  product: string;
+}
+
+export const PRODUCT_ID = 1;
+
+const databaseUrl =
```

Comment:
Should we use the same logic for configuring the db connection in the app and the tests?